app/views/hosts/multiple_disassociate.html.erb

```diff
@@ -1,5 +1,12 @@
-<%= render 'selected_hosts', :hosts => @hosts %>
+<% non_physical_hosts = @hosts.with_compute_resource %>
```

Comment:
This partial contains a lot of logic - can you extract it to a helper and test it?